loader_hub/mangoapps_guides/base.py

```diff
@@ -0,0 +1,186 @@
+"""MangoppsGuides reader."""
+import json
+from typing import List
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+from bs4 import BeautifulSoup
+from concurrent.futures import ThreadPoolExecutor
+import requests
+from urllib.parse import urlparse
+
+
+class MangoppsGuidesReader(BaseReader):
+    """MangoppsGuides reader. Reads data from a MangoppsGuides workspace.
+
+    Args:
+        domain_url (str): MangoppsGuides domain url
+        limir (int): depth to crawl
+    """
+
+    def __init__(self, domain_url: str, limit: int) -> None:
```

Comment:
seems like domain_url and limit can be parameters in load_data, not as `__init__` (i would think of initialization params as those that would be common to all requests)

Reply:
thanks, it is done now